plugins/catalog-backend/src/search/Collator.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  IndexableDocument,
+  IndexableDocumentCollator,
+} from '@backstage/plugin-search-indexer-backend';
+import { EntitiesCatalog } from '../catalog';
+import { EntityFilters } from '../service/EntityFilters';
+
+export interface CatalogEntityDocument extends IndexableDocument {
+  componentType: string;
+}
+
+export const SearchCollatorFactory = (
+  entitiesCatalog: EntitiesCatalog,
+): IndexableDocumentCollator => {
+  return async (): Promise<IndexableDocument[]> => {
+    const filter = EntityFilters.ofQuery({ kind: 'Component' });
```

Comment:
Perhaps a bit strange to only query for components here, but something we can iterate on too